template.rb

```diff
@@ -35,6 +35,112 @@
 bundle exec rspec
 ```
 
+## PR Workflow
+
+### Commits
+
+Commits should have a descriptive subject as well as a quick explanation of the reason for the change in the commit
+body.
+This makes it easier to check changes in the code editor as you do not have to find the pull request and open it
+on github.
+These commit bodies can also be used to fill the content of the pull request if you wish.
+
+### Branches
+
+Branches should be opened from the master branch. Naming convention is {type}/{task-number}{descriptive-task-name}
+
+#### Types:
+
+- feature
+
+  A new feature or an improvement to an existing feature.
+
+- fix
+
+  A non-critical bugfix, improvement, paying the technical debt. Goes through code review process.
+
+- hotfix
+
+  A time sensitive critical bugfix, that should be deployed to production as soon as possible. Not necessary that it
+  goes through code review, but it should be revisited at a later stage, and properly fixed or improved.
+
+### Pull Requests
+
+Once the feature or fix is done, a PR for it should be opened. We have a pull request template with placeholders for all
+relevant data that should be included. Code-owners are automatically assigned as reviewers.
+
+#### Labels
+
+We're using labels on PRs to visually mark the different states of the PRs. Some are self-explanatory, others have an
+additional description on github.
+
+[TODO] Add list and descriptions of project-specific labels here.
+
+#### Solving Change Requests
+
+Change requests should be fixed in separate fixup or squash commits. Rebasing the branch during an ongoing review is not
+appreciated unless there is a good reason for it, like pulling in some new and necessary changes from master, because it
+makes harder for the reviewers to know what the new changes are and what they already reviewed.
+
+### Staging
+
+Once the PR is opened, the feature can be merged into staging,
+unless it contains considerable logic in the migrations,
+in which case the reviewers should prioritize reviewing the migrations first
+and giving a thumbs up for a merge into staging.
+
+##### Merge Methodology
```

Comment:
The industry-adopted term for this procedure is "integration" and it covers any type of "joining" features, merging, rebasing, cherry-picking, ...
```suggestion
##### Integration Methodology
```